### DISCUSSION TITLE
Is there a pure Python version of `pydantic-core`?

### DISCUSSION THREAD

**Discussion by tyteen4a03** [pydantic/pydantic#10859] [Pydantic V2]
Because of my custom build environment, it's not possible for me to use any C or Rust extensions in my application code, which blocks me from using Pydantic v2 as `pydantic-core` is written in Rust.

Is there a pure Python version of `pydantic-core` that I can install instead?

**Comment by Viicos** [comment]
There isn't, and I don't think this will happen considering the amount of work it would require.

**Comment by barracuda156** [comment]
@tyteen4a03 Unfortunately, until someone manages to rewrite v2 back to python, the only working solution is to use pydantic v1. That may require some modification to its dependencies, if those lack v1-compatible code.
